packages/apollo-cli/src/generate.ts

```diff
@@ -118,6 +115,52 @@ export default function generate(
         generatedFiles.map(o => o.content.fileContents).join("\n") + common
       );
 
+      writtenFiles += 1;
+    }
+  } else if (target === "typescript" || target === "ts") {
+    const context = compileToIR(schema, document, options);
+    const generatedFiles = generateTypescriptLocalSource(context);
+    const generatedGlobalFile = generateTypescriptGlobalSource(context);
+
+    const outFiles: {
+      [fileName: string]: BasicGeneratedFile;
+    } = {};
+
+    if (nextToSources || (fs.existsSync(outputPath) && fs.statSync(outputPath).isDirectory())) {
```

Comment:
This logic still seems pretty similar to Flow other than the change in the call to `content`. Would it be possible to extract out the common logic?

Reply:
In fact, you could probably use the TypeScript code for Flow as is. The Flow generator would receive the additional property `globalSourcePath` but it would just ignore it.

Reply:
So the separation was with intent. Yes there is a lot of logic, but there is also https://github.com/apollographql/apollo-cli/pull/410 which I didn't want to interfere with. I wanted to keep them separate until Flow catches up and dedups as well. Then those paths could be merged again (if similar enough). What do you think?

If we use the TS code as is for flow, we would also have to modify the flow package to return content as a closure. (Which I am not really interested in doing to be honest as I don't care much about Flow.)